server/src/main/java/org/candlepin/dto/api/v1/EnvironmentDTO.java

```diff
@@ -0,0 +1,276 @@
+    }
+
+    /**
+     * Sets the owner to set on this EnvironmentDTO object.
+     *
+     * @param owner the owner to set on this EnvironmentDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public EnvironmentDTO setOwner(OwnerDTO owner) {
+        this.owner = owner;
+        return this;
+    }
+
+    /**
+     * Retrieves the environmentContent field of this EnvironmentDTO object.
+     *
+     * @return the environmentContent field of this EnvironmentDTO object.
+     */
+    public Set<EnvironmentContentDTO> getEnvironmentContent() {
+        return this.environmentContent != null ?
+            new SetView<EnvironmentContentDTO>(this.environmentContent) : null;
+    }
+
+    /**
+     * Sets the environmentCOntent to set on this EnvironmentDTO object.
+     *
+     * @param environmentContent the environementContent to set on this EnvironmentDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public EnvironmentDTO setEnvironmentContent(Set<EnvironmentContentDTO> environmentContent) {
+        if (environmentContent != null) {
+            if (this.environmentContent == null) {
+                this.environmentContent = new HashSet<EnvironmentContentDTO>();
+            }
+            else {
+                this.environmentContent.clear();
+            }
+
+            for (EnvironmentContentDTO dto : environmentContent) {
+                if (dto == null || dto.getContent() == null ||
+                    StringUtils.isEmpty(dto.getContent().getId())) {
+                    throw new IllegalArgumentException("environment content is null or incomplete");
+                }
+
+                this.environmentContent.add(new EnvironmentContentDTO(dto));
+            }
+        }
+        else {
+            this.environmentContent = null;
+        }
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        return String.format(
+            "EnvironmentDTO [id: %s, name: %s]",
+            this.getId(), this.getName());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj instanceof EnvironmentDTO && super.equals(obj)) {
+            EnvironmentDTO that = (EnvironmentDTO) obj;
+
+            String thisOid = this.getOwner() != null ? this.getOwner().getId() : null;
+            String thatOid = that.getOwner() != null ? that.getOwner().getId() : null;
+
+            EqualsBuilder builder = new EqualsBuilder()
+                .append(this.getId(), that.getId())
+                .append(this.getName(), that.getName())
+                .append(this.getDescription(), that.getDescription())
+                .append(thisOid, thatOid);
+
+            return builder.isEquals() &&
+                Util.collectionsAreEqual(this.getEnvironmentContent(), that.getEnvironmentContent());
```

Comment:
@nikosmoum I kept the implementation similar to ProductDTO equals, which does similar computation on content. thoughts?

Reply:
crog mentioned this on the PoolDTO so I thought that's the way we do it. Not sure if if ProductDTO is an exception for some reason, or if it was missed. @Ceiu ^?

Reply:
ProductDTO uses the collectionsAreEqual on the sets because they're sets of ProductContentDTO objects, which themselves do the ID comparison. 

As an aside, we're looking at the UUIDs in ProductContentDTO, which might be wrong. It probably needs to look at UUIDs if they exist, and then fall back to IDs. More thought is needed there, I think.

Reply:
Actually, I just noticed that I implemented that suggestion wrongly on my PR :/